scraper/src/github-scraper/parseEvents.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  Activity,
+  IGitHubEvent,
+  ProcessData,
+  PullRequestEvent,
+} from "./types.js";
+import { calculateTurnaroundTime } from "./utils.js";
+import { parseISO } from "date-fns";
+import { isBlacklisted } from "./utils.js";
+import { octokit } from "./config.js";
+
+let processedData: ProcessData = {};
+
+function appendEvent(user: string, event: Activity) {
+  console.log(`Appending event for ${user}`);
+  if (!processedData[user]) {
+    console.log(`Creating new user data for ${user}`);
+    processedData[user] = {
+      last_updated: event.time,
+      activity: [event],
+      open_prs: [],
+      authored_issue_and_pr: [],
+      discussions: [],
+    };
+  } else {
+    processedData[user]["activity"].push(event);
+    if (event["time"] > (processedData[user]["last_updated"] ?? 0)) {
+      processedData[user]["last_updated"] = event["time"];
+    }
+  }
+}
+
+async function addCollaborations(event: PullRequestEvent, eventTime: Date) {
+  let nameUserCache: { [key: string]: string } = {};
+  let emailUserCache: { [key: string]: string } = {};
```

Comment:
Are you sure the name and email user cache will work as intended here? Wouldn't the objects be empty everytime the function is called with a new PR event?